### ISSUE TITLE
On ubuntu 18.04 - installing curl removes .NET Runtime package

### ISSUE THREAD

**Issue by rakeshsinghranchi** [dotnet/core-setup#3875]
Work for producing native .NET Core installer for 2.0.0 is done and enabled in daily build in release/2.0.0 branch . During testing it is found that curl has dependency on libcurl4 that removes libcurl3 and openssl1.0 and thus, dotnet-runtime-2.0.7 
## Steps to reproduce
On docker image ( Ubuntu:18.04 ) 
--create container
docker run --rm -it Ubuntu:18.04 
-- inside container
apt-get update
apt-get install wget
--download latest 2.0.x packages  to a directory
mkdir downloadpackages && cd downloadpackages
wget https://dotnetcli.blob.core.windows.net/dotnet/Runtime/release/2.0.0/dotnet-host-latest-ubuntu.18.04-x64.deb
wget https://dotnetcli.blob.core.windows.net/dotnet/Runtime/release/2.0.0/dotnet-hostfxr-latest-ubuntu.18.04-x64.deb
wget https://dotnetcli.blob.core.windows.net/dotnet/Runtime/release/2.0.0/dotnet-runtime-latest-ubuntu.18.04-x64.deb

--install packages:
dpkg -I *.deb
apt-get install -f

-- install curl
apt-get install curl

## Expected  behavior
curl installs successfully

## Actual behavior
root@5dbf616ad5f7:/downloadpackages# apt-get install curl
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libfreetype6 libglib2.0-0 libglib2.0-data libgraphite2-3 libharfbuzz0b libicu-le-hb0 libicu60 liblttng-ust-ctl4 liblttng-ust0 libpng16-16 libssl1.0.0 libunwind8 liburcu6 libxml2 shared-mime-info xdg-user-dirs
Use 'apt autoremove' to remove them.
The following additional packages will be installed:
  libcurl4
The following packages will be REMOVED:
  dotnet-runtime-2.0.7-servicing-26322-01 **libcurl3**
The following NEW packages will be installed:
  curl libcurl4

## Environment data

NAME="Ubuntu"
VERSION="18.04 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu Bionic Beaver (development branch)"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

**Comment by rakeshsinghranchi** [comment]
cc @leecow 
@janvorli  , note that this issue is opened for .NET 2.0.x and I believe will exist across the releases.

**Comment by janvorli** [comment]
For 2.0, the libcurl4 works fine. Openssl1.0.0 can be installed side by side it with no problem.

**Comment by janvorli** [comment]
For 2.1 too.

**Comment by rakeshsinghranchi** [comment]
Looks like Debian package tool determines libcurl3 as the dependency instead of libcurl4 . On Ubuntu 18.04, libcurl3 and libcurl4 are not SxS

**Comment by borgdylan** [comment]
So does cmake. Any .NET Core runtime package I have from the 1.0 days till today will get removed If I try installing cmake. This means I cannot build mono or .NET Core on my machine. I have already commented on a issue on launchpad asking for a fix from canonical.

**Comment by stephentoub** [comment]
I'm not clear on why .NET Core 1.x packages depended on libcurl3 instead of libcurl4.  From a source perspective, it's always expected to work with 4:
https://github.com/dotnet/corefx/blob/bd0d24e5567bc789b82a4ead455ad43634676372/src/Native/System.Net.Http.Native/CMakeLists.txt#L5

**Comment by borgdylan** [comment]
The metadata in the deb packages asked for libcurl3 not libcurl4. This might have been to libcurl4 being unavailable in the past. Libraries get upgraded on a 6 month cadence by canonical, forcing everyone to rebuild their packages. One should not be expected to not upgrade because of .NET Core only.

**Comment by Petermarcu** [comment]
We'll need to get the packages updated that we put on the 18.04 repo to express a libcurl4 dependency.

**Comment by borgdylan** [comment]
Also, any uploads to sais repo should happen within a week from now. Ubuntu 18.04 RTMs next Thursday.

**Comment by ianhays** [comment]
The libcurl issue doesn’t appear to be present for 2.1 and master, only 2.0.

I pulled down the latest master build and checked the deb file imports and it doesn’t list libcurl:
Package: dotnet-runtime-deps-2.2.0-preview1-26420-14
Version: 2.2.0-preview1-26420-14-1
Depends: libgcc1, libssl1.0.0, liblttng-ust0, libc6, libicu60, zlib1g, libstdc++6, libgssapi-krb5-2

Which is [to be expected looking at the source](https://github.com/dotnet/core-setup/blob/master/src/pkg/packaging/deb/dotnet-runtime-deps-debian_config_ubuntu.18.04-x64.json#L31)

So we only have an issue for Ubuntu1804 in 2.0, which does contain libcurl3 in the deb imports list:

Package: dotnet-runtime-2.0.8-servicing-26407-02
Version: 2.0.8-servicing-26407-02-1
Depends: libc6 (>= 2.14), libcurl3 (>= 7.16.2), libgcc1 (>= 1:3.0), libgssapi-krb5-2 (>= 1.14+dfsg), liblttng-ust0 (>= 2.5.0), libstdc++6 (>= 4.8), libunwind8, libuuid1 (>= 2.16), zlib1g (>= 1:1.1.4), libssl1.0.0, libicu60, dotnet-hostfxr-2.0.8-servicing-26407-02

Jeremy and I are looking into fixing rel2.0 now. I’m trying to determine where the libcurl3 dependency is coming from, because the closet correlary to the 2.1 runtime deps config file I could find doesn't list libcurl

**Comment by borgdylan** [comment]
The issue seems to impack anything using .NET Core 2.0 like powershell and also unrelated stuff like mssql-tools. How can I get a message to all of microsoft that new packages for Ubuntu 18.04 need to issued ASAP?

**Comment by borgdylan** [comment]
Manually the technique shown here can be used to modify an already built package: https://bugs.launchpad.net/ubuntu/+source/curl/+bug/1754294/comments/25

**Comment by leecow** [comment]
libcurl dependency fixed up in the 2.0.7 packages which are now available on 18.04 feed. If you haven't already worked out where the feed is hiding, this will get you set up. 

```
wget -qO- https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > microsoft.asc.gpg
mv microsoft.asc.gpg /etc/apt/trusted.gpg.d/
wget -q https://packages.microsoft.com/config/ubuntu/18.04/prod.list
mv prod.list /etc/apt/sources.list.d/microsoft-prod.list
```

**Comment by dagood** [comment]
Closing: stale issue. This is fixed in current versions of the packages.
